File: 프로그래머스/lv2/42746. 가장 큰 수/가장 큰 수.py
from functools import cmp_to_key


def solution(numbers):
    strs = list(map(str, numbers))
    strs.sort(key=cmp_to_key(compare), reverse=True)
    return str(int(''.join(strs)))


def compare(a, b):
    return int(a + b) - int(b + a)

File: 프로그래머스/lv2/42746. 가장 큰 수/가장 큰 수.kt
class Solution {
    fun solution(numbers: IntArray): String {
        val strs = numbers.map { it.toString() }
        val sorted_strs = strs.sortedWith { a, b -> (b + a).toInt() - (a + b).toInt() }
        return sorted_strs.joinToString("").let {
            if (it.startsWith("0")) "0" else it
        }
    }
}